File: apk/app/src/main/java/com/example/momentra/ui/shell/business/BusinessActiveTheme.kt
package com.example.momentra.ui.shell.business

import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import com.example.momentra.ui.shell.empty.BusinessSetupKind

/** B01–B03 Business active tokens — Team Ops / Runway / Operations. */
data class BusinessActiveTheme(
    val bg: Color,
    val accent: Color,
    val accentSolid: Color,
    val accentLight: Color,
    val accentSoft: Color,
    val text: Color,
    val secondary: Color,
    val muted: Color,
    val card: Color,
    val border: Color,
    val typeLabel: String,
    val pulseTitle: String,
    val momentsTitle: String,
    val lifeTitle: String,
    val memoryTitle: String,
    val hubSubtitle: String,
    val hubHeroTitle: String,
    val hubHeroDetail: String,
    val hubHeroRes: Int,
    val filterChips: List<String>,
    val heroGradient: Brush,
) {
    companion object {
        val TeamOperations = BusinessActiveTheme(
            bg = Color(0xFF0C0F15),
            accent = Color(0xFF10B981),
            accentSolid = Color(0xFF059669),
            accentLight = Color(0xFF34D399),
            accentSoft = Color(0x3310B981),
            text = Color(0xFFE5E0EE),
            secondary = Color(0xFF94A3B8),
            muted = Color(0xFF64748B),
            card = Color(0xFF161B26),
            border = Color(0xFF1E293B),
            typeLabel = "Team Operations",
            pulseTitle = "Team Pulse",
            momentsTitle = "Team Moments",
            lifeTitle = "Team Life",
            memoryTitle = "Team Memory",
            hubSubtitle = "Bring your team operations to life",
            hubHeroTitle = "Bring your team operations to life",
            hubHeroDetail = "Add people, plans, decisions, and updates",
            hubHeroRes = com.example.momentra.R.drawable.team_ops_hub_hero,
            filterChips = listOf("Team Sync", "Sprint Review"),
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFF6366F1), Color(0xFFA855F7))),
        )

        val BusinessRunway = BusinessActiveTheme(
            bg = Color(0xFF0C0F15),
            accent = Color(0xFFF59E0B),
            accentSolid = Color(0xFFD97706),
            accentLight = Color(0xFFFBBF24),
            accentSoft = Color(0x33F59E0B),
            text = Color(0xFFE5E0EE),
            secondary = Color(0xFF94A3B8),
            muted = Color(0xFF64748B),
            card = Color(0xFF161B26),
            border = Color(0xFF1E293B),
            typeLabel = "Business Runway",
            pulseTitle = "Runway Pulse",
            momentsTitle = "Runway Moments",
            lifeTitle = "Runway Life",
            memoryTitle = "Runway Memory",
            hubSubtitle = "Bring your finances to life",
            hubHeroTitle = "Bring your finances to life",
            hubHeroDetail = "Track revenue, expenses, taxes and forecasts",
            hubHeroRes = com.example.momentra.R.drawable.runway_hub_hero,
            filterChips = listOf("Revenue", "Expenses"),
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFF6366F1), Color(0xFFA855F7))),
        )

        val BusinessOperations = BusinessActiveTheme(
            bg = Color(0xFF0C0F15),
            accent = Color(0xFF818CF8),
            accentSolid = Color(0xFF6366F1),
            accentLight = Color(0xFFA5B4FC),
            accentSoft = Color(0x33818CF8),
            text = Color(0xFFE5E0EE),
            secondary = Color(0xFF94A3B8),
            muted = Color(0xFF64748B),
            card = Color(0xFF161B26),
            border = Color(0xFF1E293B),
            typeLabel = "Business Operations",
            pulseTitle = "Ops Pulse",
            momentsTitle = "Ops Moments",
            lifeTitle = "Ops Life",
            memoryTitle = "Ops Memory",
            hubSubtitle = "Bring your operations to life",
            hubHeroTitle = "Bring your operations to life",
            hubHeroDetail = "Add expenses, vendors, approvals and updates",
            hubHeroRes = com.example.momentra.R.drawable.ops_hub_hero,
            filterChips = listOf("Budget Ops", "Vendor Mgmt"),
            heroGradient = Brush.horizontalGradient(listOf(Color(0xFF6366F1), Color(0xFFA855F7))),
        )

        fun forKind(kind: BusinessSetupKind): BusinessActiveTheme = when (kind) {
            BusinessSetupKind.TEAM_OPERATIONS -> TeamOperations
            BusinessSetupKind.BUSINESS_RUNWAY -> BusinessRunway
            BusinessSetupKind.BUSINESS_OPERATIONS -> BusinessOperations
        }

        fun forTypeCode(momentTypeCode: String?): BusinessActiveTheme {
            val code = momentTypeCode?.uppercase().orEmpty()
            return when {
                code.contains("RUNWAY") -> BusinessRunway
                code.contains("OPERATIONS") && !code.contains("TEAM") -> BusinessOperations
                else -> TeamOperations
            }
        }
    }
}

enum class BusinessQuickAddKind {
    TEAM_UPDATE,
    DECISION,
    BLOCKER,
    MEETING,
    RECOGNITION,
    APPROVAL,
    MILESTONE,
    RETROSPECTIVE,
    RISK_FLAG,
    ACTIVITY_LOG,
    POLL,
    MEMORY,
    REVENUE,
    EXPENSE,
    TAX_ENTRY,
    INVESTOR_UPDATE,
    BUDGET_ALERT,
    FORECAST_UPDATE,
    INVOICE,
    GENERAL_UPDATE,
    SPEND_ENTRY,
    UPDATE_VENDOR,
    REQUEST_APPROVAL,
    REPORT_ISSUE,
    LOG_IMPROVEMENT,
    BUDGET_REVIEW,
    SLA_CHECK,
}

fun BusinessQuickAddKind.isLive(): Boolean = when (this) {
    BusinessQuickAddKind.EXPENSE,
    BusinessQuickAddKind.SPEND_ENTRY,
    BusinessQuickAddKind.REVENUE,
    BusinessQuickAddKind.INVOICE,
    BusinessQuickAddKind.TAX_ENTRY,
    BusinessQuickAddKind.INVESTOR_UPDATE,
    BusinessQuickAddKind.BUDGET_ALERT,
    BusinessQuickAddKind.FORECAST_UPDATE,
    BusinessQuickAddKind.POLL,
    BusinessQuickAddKind.MEMORY,
    BusinessQuickAddKind.TEAM_UPDATE,
    BusinessQuickAddKind.DECISION,
    BusinessQuickAddKind.BLOCKER,
    BusinessQuickAddKind.MEETING,
    BusinessQuickAddKind.RECOGNITION,
    BusinessQuickAddKind.APPROVAL,
    BusinessQuickAddKind.MILESTONE,
    BusinessQuickAddKind.RETROSPECTIVE,
    BusinessQuickAddKind.RISK_FLAG,
    BusinessQuickAddKind.ACTIVITY_LOG,
    BusinessQuickAddKind.GENERAL_UPDATE,
    BusinessQuickAddKind.UPDATE_VENDOR,
    BusinessQuickAddKind.REQUEST_APPROVAL,
    BusinessQuickAddKind.REPORT_ISSUE,
    BusinessQuickAddKind.LOG_IMPROVEMENT,
    BusinessQuickAddKind.BUDGET_REVIEW,
    BusinessQuickAddKind.SLA_CHECK,
    -> true
    else -> false
}

fun BusinessQuickAddKind.label(): String = when (this) {
    BusinessQuickAddKind.TEAM_UPDATE -> "Team Update"
    BusinessQuickAddKind.DECISION -> "Decision"
    BusinessQuickAddKind.BLOCKER -> "Blocker"
    BusinessQuickAddKind.MEETING -> "Meeting"
    BusinessQuickAddKind.RECOGNITION -> "Recognition"
    BusinessQuickAddKind.APPROVAL -> "Approval"
    BusinessQuickAddKind.MILESTONE -> "Milestone"
    BusinessQuickAddKind.RETROSPECTIVE -> "Retrospective"
    BusinessQuickAddKind.RISK_FLAG -> "Risk Flag"
    BusinessQuickAddKind.ACTIVITY_LOG -> "Activity Log"
    BusinessQuickAddKind.POLL -> "Poll"
    BusinessQuickAddKind.MEMORY -> "Memory"
    BusinessQuickAddKind.REVENUE -> "Log Revenue"
    BusinessQuickAddKind.EXPENSE -> "Log Expense"
    BusinessQuickAddKind.TAX_ENTRY -> "Tax Entry"
    BusinessQuickAddKind.INVESTOR_UPDATE -> "Investor Update"
    BusinessQuickAddKind.BUDGET_ALERT -> "Budget Alert"
    BusinessQuickAddKind.FORECAST_UPDATE -> "Forecast Update"
    BusinessQuickAddKind.INVOICE -> "Invoice Track"
    BusinessQuickAddKind.GENERAL_UPDATE -> "General Update"
    BusinessQuickAddKind.SPEND_ENTRY -> "Log Spend Entry"
    BusinessQuickAddKind.UPDATE_VENDOR -> "Update Vendor"
    BusinessQuickAddKind.REQUEST_APPROVAL -> "Request Approval"
    BusinessQuickAddKind.REPORT_ISSUE -> "Report Issue"
    BusinessQuickAddKind.LOG_IMPROVEMENT -> "Log Improvement"
    BusinessQuickAddKind.BUDGET_REVIEW -> "Budget Review"
    BusinessQuickAddKind.SLA_CHECK -> "SLA Check"
}

fun BusinessQuickAddKind.subtitle(): String = when (this) {
    BusinessQuickAddKind.TEAM_UPDATE -> "Share progress"
    BusinessQuickAddKind.DECISION -> "Log choices"
    BusinessQuickAddKind.BLOCKER -> "Flag issues"
    BusinessQuickAddKind.MEETING -> "Capture notes"
    BusinessQuickAddKind.RECOGNITION -> "Celebrate wins"
    BusinessQuickAddKind.APPROVAL -> "Route requests"
    BusinessQuickAddKind.MILESTONE -> "Mark progress"
    BusinessQuickAddKind.RETROSPECTIVE -> "Review & learn"
    BusinessQuickAddKind.RISK_FLAG -> "Raise concerns"
    BusinessQuickAddKind.ACTIVITY_LOG -> "Track actions"
    BusinessQuickAddKind.POLL -> "Gather input"
    BusinessQuickAddKind.MEMORY -> "Save learnings"
    BusinessQuickAddKind.REVENUE -> "Track income"
    BusinessQuickAddKind.EXPENSE -> "Record spend"
    BusinessQuickAddKind.TAX_ENTRY -> "File taxes"
    BusinessQuickAddKind.INVESTOR_UPDATE -> "Share metrics"
    BusinessQuickAddKind.BUDGET_ALERT -> "Flag overrun"
    BusinessQuickAddKind.FORECAST_UPDATE -> "Update projections"
    BusinessQuickAddKind.INVOICE -> "Monitor payments"
    BusinessQuickAddKind.GENERAL_UPDATE -> "Share updates"
    BusinessQuickAddKind.SPEND_ENTRY -> "Record expenses"
    BusinessQuickAddKind.UPDATE_VENDOR -> "Update suppliers"
    BusinessQuickAddKind.REQUEST_APPROVAL -> "Request sign-off"
    BusinessQuickAddKind.REPORT_ISSUE -> "Flag a problem"
    BusinessQuickAddKind.LOG_IMPROVEMENT -> "Log optimization"
    BusinessQuickAddKind.BUDGET_REVIEW -> "Check budgets"
    BusinessQuickAddKind.SLA_CHECK -> "Monitor SLAs"
}

fun BusinessQuickAddKind.emoji(): String = when (this) {
    BusinessQuickAddKind.TEAM_UPDATE, BusinessQuickAddKind.GENERAL_UPDATE -> "📢"
    BusinessQuickAddKind.DECISION -> "🚩"
    BusinessQuickAddKind.BLOCKER, BusinessQuickAddKind.RISK_FLAG, BusinessQuickAddKind.REPORT_ISSUE -> "🛡"
    BusinessQuickAddKind.MEETING, BusinessQuickAddKind.BUDGET_REVIEW -> "📅"
    BusinessQuickAddKind.RECOGNITION -> "⭐"
    BusinessQuickAddKind.APPROVAL, BusinessQuickAddKind.REQUEST_APPROVAL -> "🏪"
    BusinessQuickAddKind.MILESTONE -> "🏁"
    BusinessQuickAddKind.RETROSPECTIVE -> "⚡"
    BusinessQuickAddKind.ACTIVITY_LOG -> "📈"
    BusinessQuickAddKind.POLL -> "📊"
    BusinessQuickAddKind.MEMORY -> "📦"
    BusinessQuickAddKind.REVENUE -> "💰"
    BusinessQuickAddKind.EXPENSE, BusinessQuickAddKind.SPEND_ENTRY -> "💳"
    BusinessQuickAddKind.TAX_ENTRY -> "🧾"
    BusinessQuickAddKind.INVESTOR_UPDATE -> "📣"
    BusinessQuickAddKind.BUDGET_ALERT -> "🚨"
    BusinessQuickAddKind.FORECAST_UPDATE -> "📉"
    BusinessQuickAddKind.INVOICE -> "📄"
    BusinessQuickAddKind.UPDATE_VENDOR -> "🏷"
    BusinessQuickAddKind.LOG_IMPROVEMENT -> "✨"
    BusinessQuickAddKind.SLA_CHECK -> "⏱"
}

fun BusinessQuickAddKind.stripeColor(): Color = when (this) {
    BusinessQuickAddKind.TEAM_UPDATE, BusinessQuickAddKind.GENERAL_UPDATE,
    BusinessQuickAddKind.APPROVAL, BusinessQuickAddKind.ACTIVITY_LOG, BusinessQuickAddKind.SPEND_ENTRY,
    BusinessQuickAddKind.BUDGET_REVIEW,
    -> Color(0xFF818CF8)
    BusinessQuickAddKind.DECISION, BusinessQuickAddKind.INVESTOR_UPDATE,
    BusinessQuickAddKind.UPDATE_VENDOR, BusinessQuickAddKind.MEMORY,
    -> Color(0xFFA78BFA)
    BusinessQuickAddKind.BLOCKER, BusinessQuickAddKind.RISK_FLAG, BusinessQuickAddKind.BUDGET_ALERT,
    BusinessQuickAddKind.REPORT_ISSUE,
    -> Color(0xFFEF4444)
    BusinessQuickAddKind.MEETING, BusinessQuickAddKind.REVENUE, BusinessQuickAddKind.EXPENSE,
    BusinessQuickAddKind.FORECAST_UPDATE, BusinessQuickAddKind.REQUEST_APPROVAL,
    BusinessQuickAddKind.RETROSPECTIVE,
    -> Color(0xFFF59E0B)
    BusinessQuickAddKind.RECOGNITION, BusinessQuickAddKind.TAX_ENTRY, BusinessQuickAddKind.LOG_IMPROVEMENT,
    BusinessQuickAddKind.POLL,
    -> Color(0xFF10B981)
    BusinessQuickAddKind.MILESTONE, BusinessQuickAddKind.INVOICE, BusinessQuickAddKind.SLA_CHECK -> Color(0xFF14B8A6)
}

/** Figma 649:26162 Team Ops Action Center tile icons (MCP exports). */
fun BusinessQuickAddKind.teamOpsHubIconRes(): Int? = when (this) {
    BusinessQuickAddKind.TEAM_UPDATE -> com.example.momentra.R.drawable.ic_teamops_qa_update
    BusinessQuickAddKind.DECISION -> com.example.momentra.R.drawable.ic_teamops_qa_decision
    BusinessQuickAddKind.BLOCKER -> com.example.momentra.R.drawable.ic_teamops_qa_blocker
    BusinessQuickAddKind.MEETING -> com.example.momentra.R.drawable.ic_teamops_qa_meeting
    BusinessQuickAddKind.RECOGNITION -> com.example.momentra.R.drawable.ic_teamops_qa_recognition
    BusinessQuickAddKind.APPROVAL -> com.example.momentra.R.drawable.ic_teamops_qa_approval
    BusinessQuickAddKind.MILESTONE -> com.example.momentra.R.drawable.ic_teamops_qa_milestone
    BusinessQuickAddKind.RETROSPECTIVE -> com.example.momentra.R.drawable.ic_teamops_qa_retro
    BusinessQuickAddKind.RISK_FLAG -> com.example.momentra.R.drawable.ic_teamops_qa_risk
    BusinessQuickAddKind.ACTIVITY_LOG -> com.example.momentra.R.drawable.ic_teamops_qa_activity
    BusinessQuickAddKind.POLL -> com.example.momentra.R.drawable.ic_teamops_qa_poll
    BusinessQuickAddKind.MEMORY -> com.example.momentra.R.drawable.ic_teamops_qa_memory
    else -> null
}

fun businessHubTiles(theme: BusinessActiveTheme): List<BusinessQuickAddKind> = when (theme.typeLabel) {
    "Business Runway" -> listOf(
        BusinessQuickAddKind.REVENUE,
        BusinessQuickAddKind.EXPENSE,
        BusinessQuickAddKind.TAX_ENTRY,
        BusinessQuickAddKind.INVESTOR_UPDATE,
        BusinessQuickAddKind.BUDGET_ALERT,
        BusinessQuickAddKind.FORECAST_UPDATE,
        BusinessQuickAddKind.INVOICE,
        BusinessQuickAddKind.GENERAL_UPDATE,
        BusinessQuickAddKind.MEMORY,
    )
    "Business Operations" -> listOf(
        BusinessQuickAddKind.SPEND_ENTRY,
        BusinessQuickAddKind.UPDATE_VENDOR,
        BusinessQuickAddKind.REQUEST_APPROVAL,
        BusinessQuickAddKind.REPORT_ISSUE,
        BusinessQuickAddKind.LOG_IMPROVEMENT,
        BusinessQuickAddKind.BUDGET_REVIEW,
        BusinessQuickAddKind.SLA_CHECK,
        BusinessQuickAddKind.GENERAL_UPDATE,
        BusinessQuickAddKind.MEMORY,
    )
    else -> listOf(
        BusinessQuickAddKind.TEAM_UPDATE,
        BusinessQuickAddKind.DECISION,
        BusinessQuickAddKind.BLOCKER,
        BusinessQuickAddKind.MEETING,
        BusinessQuickAddKind.RECOGNITION,
        BusinessQuickAddKind.APPROVAL,
        BusinessQuickAddKind.MILESTONE,
        BusinessQuickAddKind.RETROSPECTIVE,
        BusinessQuickAddKind.RISK_FLAG,
        BusinessQuickAddKind.ACTIVITY_LOG,
        BusinessQuickAddKind.POLL,
        BusinessQuickAddKind.MEMORY,
    )
}

File: momentra/momentra/Shell/BusinessActive/Shared/BusinessActiveTheme.swift
import SwiftUI

/// B01–B03 Business active tokens — Team Ops / Runway / Operations.
struct BusinessActiveTheme {
    let bg: Color
    let accent: Color
    let accentSolid: Color
    let accentLight: Color
    let accentSoft: Color
    let text: Color
    let secondary: Color
    let muted: Color
    let card: Color
    let border: Color
    let typeLabel: String
    let pulseTitle: String
    let momentsTitle: String
    let lifeTitle: String
    let memoryTitle: String
    let hubSubtitle: String
    let hubHeroTitle: String
    let hubHeroDetail: String
    let hubHeroAssetName: String
    let filterChips: [String]
    let heroGradientColors: [Color]
    let quickChips: [(emoji: String, label: String, kind: BusinessQuickAddKind)]

    var heroGradient: LinearGradient {
        LinearGradient(colors: heroGradientColors, startPoint: .leading, endPoint: .trailing)
    }

    static let teamOperations = BusinessActiveTheme(
        bg: Color(hex: "#0C0F15"),
        accent: Color(hex: "#10B981"),
        accentSolid: Color(hex: "#059669"),
        accentLight: Color(hex: "#34D399"),
        accentSoft: Color(hex: "#10B981").opacity(0.2),
        text: Color(hex: "#E5E0EE"),
        secondary: Color(hex: "#94A3B8"),
        muted: Color(hex: "#64748B"),
        card: Color(hex: "#161B26"),
        border: Color(hex: "#1E293B"),
        typeLabel: "Team Operations",
        pulseTitle: "Team Pulse",
        momentsTitle: "Team Moments",
        lifeTitle: "Team Life",
        memoryTitle: "Team Memory",
        hubSubtitle: "Bring your team operations to life",
        hubHeroTitle: "Bring your team operations to life",
        hubHeroDetail: "Add people, plans, decisions, and updates",
        hubHeroAssetName: "TeamOpsHubHero",
        filterChips: ["Team Sync", "Sprint Review"],
        heroGradientColors: [Color(hex: "#6366F1"), Color(hex: "#A855F7")],
        quickChips: [
            ("📢", "Update", .teamUpdate),
            ("🚩", "Decision", .decision),
            ("🛡", "Blocker", .blocker),
            ("📷", "Memory", .memory),
        ]
    )

    static let businessRunway = BusinessActiveTheme(
        bg: Color(hex: "#0C0F15"),
        accent: Color(hex: "#F59E0B"),
        accentSolid: Color(hex: "#D97706"),
        accentLight: Color(hex: "#FBBF24"),
        accentSoft: Color(hex: "#F59E0B").opacity(0.2),
        text: Color(hex: "#E5E0EE"),
        secondary: Color(hex: "#94A3B8"),
        muted: Color(hex: "#64748B"),
        card: Color(hex: "#161B26"),
        border: Color(hex: "#1E293B"),
        typeLabel: "Business Runway",
        pulseTitle: "Runway Pulse",
        momentsTitle: "Runway Moments",
        lifeTitle: "Runway Life",
        memoryTitle: "Runway Memory",
        hubSubtitle: "Bring your finances to life",
        hubHeroTitle: "Bring your finances to life",
        hubHeroDetail: "Track revenue, expenses, taxes and forecasts",
        hubHeroAssetName: "RunwayHubHero",
        filterChips: ["Revenue", "Expenses"],
        heroGradientColors: [Color(hex: "#6366F1"), Color(hex: "#A855F7")],
        quickChips: [
            ("💰", "Revenue", .revenue),
            ("💳", "Expense", .expense),
            ("📄", "Invoice", .invoice),
            ("📷", "Memory", .memory),
        ]
    )

    static let businessOperations = BusinessActiveTheme(
        bg: Color(hex: "#0C0F15"),
        accent: Color(hex: "#818CF8"),
        accentSolid: Color(hex: "#6366F1"),
        accentLight: Color(hex: "#A5B4FC"),
        accentSoft: Color(hex: "#818CF8").opacity(0.2),
        text: Color(hex: "#E5E0EE"),
        secondary: Color(hex: "#94A3B8"),
        muted: Color(hex: "#64748B"),
        card: Color(hex: "#161B26"),
        border: Color(hex: "#1E293B"),
        typeLabel: "Business Operations",
        pulseTitle: "Ops Pulse",
        momentsTitle: "Ops Moments",
        lifeTitle: "Ops Life",
        memoryTitle: "Ops Memory",
        hubSubtitle: "Bring your operations to life",
        hubHeroTitle: "Bring your operations to life",
        hubHeroDetail: "Add expenses, vendors, approvals and updates",
        hubHeroAssetName: "OpsHubHero",
        filterChips: ["Budget Ops", "Vendor Mgmt"],
        heroGradientColors: [Color(hex: "#6366F1"), Color(hex: "#A855F7")],
        quickChips: [
            ("💳", "Spend", .spendEntry),
            ("✅", "Approval", .requestApproval),
            ("🔧", "Issue", .reportIssue),
            ("📷", "Memory", .memory),
        ]
    )

    static func forKind(_ kind: BusinessSetupKind) -> BusinessActiveTheme {
        switch kind {
        case .teamOperations: return .teamOperations
        case .businessRunway: return .businessRunway
        case .businessOperations: return .businessOperations
        }
    }

    static func forTypeCode(_ momentTypeCode: String?) -> BusinessActiveTheme {
        let code = (momentTypeCode ?? "").uppercased()
        if code.contains("RUNWAY") { return .businessRunway }
        if code.contains("OPERATIONS") && !code.contains("TEAM") { return .businessOperations }
        if code.contains("TEAM") { return .teamOperations }
        return .teamOperations
    }
}

enum BusinessQuickAddKind: String, Identifiable, CaseIterable {
    // Team Ops
    case teamUpdate
    case decision
    case blocker
    case meeting
    case recognition
    case approval
    case milestone
    case retrospective
    case riskFlag
    case activityLog
    case poll
    case memory
    // Runway
    case revenue
    case expense
    case taxEntry
    case investorUpdate
    case budgetAlert
    case forecastUpdate
    case invoice
    case generalUpdate
    // Ops
    case spendEntry
    case updateVendor
    case requestApproval
    case reportIssue
    case logImprovement
    case budgetReview
    case slaCheck

    var id: String { rawValue }

    var isLive: Bool {
        switch self {
        case .expense, .spendEntry, .revenue, .invoice, .poll, .memory, .teamUpdate, .generalUpdate,
             .decision, .blocker, .meeting, .recognition, .approval, .milestone, .retrospective,
             .riskFlag, .activityLog,
             .taxEntry, .investorUpdate, .budgetAlert, .forecastUpdate,
             .updateVendor, .requestApproval, .reportIssue, .logImprovement, .budgetReview, .slaCheck:
            return true
        default:
            return false
        }
    }

    var label: String {
        switch self {
        case .teamUpdate: return "Team Update"
        case .decision: return "Decision"
        case .blocker: return "Blocker"
        case .meeting: return "Meeting"
        case .recognition: return "Recognition"
        case .approval: return "Approval"
        case .milestone: return "Milestone"
        case .retrospective: return "Retrospective"
        case .riskFlag: return "Risk Flag"
        case .activityLog: return "Activity Log"
        case .poll: return "Poll"
        case .memory: return "Memory"
        case .revenue: return "Log Revenue"
        case .expense: return "Log Expense"
        case .taxEntry: return "Tax Entry"
        case .investorUpdate: return "Investor Update"
        case .budgetAlert: return "Budget Alert"
        case .forecastUpdate: return "Forecast Update"
        case .invoice: return "Invoice Track"
        case .generalUpdate: return "General Update"
        case .spendEntry: return "Log Spend Entry"
        case .updateVendor: return "Update Vendor"
        case .requestApproval: return "Request Approval"
        case .reportIssue: return "Report Issue"
        case .logImprovement: return "Log Improvement"
        case .budgetReview: return "Budget Review"
        case .slaCheck: return "SLA Check"
        }
    }

    var subtitle: String {
        switch self {
        case .teamUpdate: return "Share progress"
        case .decision: return "Log choices"
        case .blocker: return "Flag issues"
        case .meeting: return "Capture notes"
        case .recognition: return "Celebrate wins"
        case .approval: return "Route requests"
        case .milestone: return "Mark progress"
        case .retrospective: return "Review & learn"
        case .riskFlag: return "Raise concerns"
        case .activityLog: return "Track actions"
        case .poll: return "Gather input"
        case .memory: return "Save learnings"
        case .revenue: return "Track income"
        case .expense: return "Record spend"
        case .taxEntry: return "File taxes"
        case .investorUpdate: return "Share metrics"
        case .budgetAlert: return "Flag overrun"
        case .forecastUpdate: return "Update projections"
        case .invoice: return "Monitor payments"
        case .generalUpdate: return "Share updates"
        case .spendEntry: return "Record expenses"
        case .updateVendor: return "Update suppliers"
        case .requestApproval: return "Request sign-off"
        case .reportIssue: return "Flag a problem"
        case .logImprovement: return "Log optimization"
        case .budgetReview: return "Check budgets"
        case .slaCheck: return "Monitor SLAs"
        }
    }

    var emoji: String {
        switch self {
        case .teamUpdate, .generalUpdate: return "📢"
        case .decision: return "🚩"
        case .blocker, .riskFlag, .reportIssue: return "🛡"
        case .meeting, .budgetReview: return "📅"
        case .recognition: return "⭐"
        case .approval, .requestApproval: return "🏪"
        case .milestone: return "🏁"
        case .retrospective: return "⚡"
        case .activityLog: return "📈"
        case .poll: return "📊"
        case .memory: return "📦"
        case .revenue: return "💰"
        case .expense, .spendEntry: return "💳"
        case .taxEntry: return "🧾"
        case .investorUpdate: return "📣"
        case .budgetAlert: return "🚨"
        case .forecastUpdate: return "📉"
        case .invoice: return "📄"
        case .updateVendor: return "🏷"
        case .logImprovement: return "✨"
        case .slaCheck: return "⏱"
        }
    }

    var stripeColor: Color {
        switch self {
        case .teamUpdate, .generalUpdate, .approval, .activityLog, .spendEntry, .budgetReview:
            return Color(hex: "#818CF8")
        case .decision, .investorUpdate, .updateVendor, .memory:
            return Color(hex: "#A78BFA")
        case .blocker, .riskFlag, .budgetAlert, .reportIssue: return Color(hex: "#EF4444")
        case .meeting, .revenue, .expense, .forecastUpdate, .requestApproval, .retrospective:
            return Color(hex: "#F59E0B")
        case .recognition, .taxEntry, .logImprovement, .poll: return Color(hex: "#10B981")
        case .milestone, .invoice, .slaCheck: return Color(hex: "#14B8A6")
        }
    }

    /// Figma 649:26162 Team Ops Action Center tile icons (MCP exports).
    var teamOpsHubIconName: String? {
        switch self {
        case .teamUpdate: return "TeamOpsQaUpdate"
        case .decision: return "TeamOpsQaDecision"
        case .blocker: return "TeamOpsQaBlocker"
        case .meeting: return "TeamOpsQaMeeting"
        case .recognition: return "TeamOpsQaRecognition"
        case .approval: return "TeamOpsQaApproval"
        case .milestone: return "TeamOpsQaMilestone"
        case .retrospective: return "TeamOpsQaRetro"
        case .riskFlag: return "TeamOpsQaRisk"
        case .activityLog: return "TeamOpsQaActivity"
        case .poll: return "TeamOpsQaPoll"
        case .memory: return "TeamOpsQaMemory"
        default: return nil
        }
    }

    static func hubTiles(theme: BusinessActiveTheme) -> [BusinessQuickAddKind] {
        switch theme.typeLabel {
        case "Business Runway":
            return [.revenue, .expense, .taxEntry, .investorUpdate, .budgetAlert, .forecastUpdate, .invoice, .generalUpdate, .memory]
        case "Business Operations":
            return [.spendEntry, .updateVendor, .requestApproval, .reportIssue, .logImprovement, .budgetReview, .slaCheck, .generalUpdate, .memory]
        default:
            return [.teamUpdate, .decision, .blocker, .meeting, .recognition, .approval, .milestone, .retrospective, .riskFlag, .activityLog, .poll, .memory]
        }
    }
}
